proxy/proxy_test.go

```diff
@@ -1247,6 +1248,77 @@ func TestFlusherImplementation(t *testing.T) {
 	}
 }
 
+func TestMeasureProxyWatch(t *testing.T) {
+	mo := metrics.Options{
+		Format:                     metrics.PrometheusKind,
+		EnableProxyRequestMetrics:  true,
+		EnableProxyResponseMetrics: true,
+	}
+	m := metrics.NewMetrics(mo)
+	defer m.Close()
+
+	doc := `*  -> latency("10ms") -> backendLatency("20ms") -> status(200) -> <shunt>`
+	tp, err := newTestProxyWithParams(doc, Params{Metrics: m})
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer tp.close()
+
+	u, _ := url.ParseRequestURI("https://www.example.org/hello")
+	r := &http.Request{URL: u, Method: "GET"}
+	w := httptest.NewRecorder()
+	tp.proxy.ServeHTTP(w, r)
+	if w.Code != http.StatusOK {
+		t.Error("wrong status", w.Code)
+	}
+
+	mh := metrics.NewHandler(mo, m)
+	w = httptest.NewRecorder()
+	r = httptest.NewRequest("GET", "/metrics", nil)
+	mh.ServeHTTP(w, r)
+	if w.Code != http.StatusOK {
+		t.Error("wrong status for metrics", w.Code)
+	}
+
+	data := make(map[string]float64)
+	for _, line := range strings.Split(w.Body.String(), "\n") {
+		if strings.HasPrefix(line, "skipper_proxy_") {
+			parts := strings.Split(line, " ")
+			if len(parts) != 2 {
+				t.Error("invalid metrics line:", line)
+				continue
+			}
+			val, err := strconv.ParseFloat(parts[1], 64)
+			if err != nil {
+				t.Errorf("invalid float value in metrics line: %v", err)
+				continue
+			}
+			data[parts[0]] = val
+		}
+	}
```

Comment:
I am not sure its a good idea to implement prometheus parsing ad-hoc here. I think we should rather implement MockMetrics.MeasureProxy and then use m.WithMeasures like in other tests.